code/WorkInProgress/nadir_antenna.dm

```diff
@@ -25,22 +49,37 @@ var/global/obj/machinery/communications_dish/transception/transception_array
 	bound_width = 96
 	mats = 0
 
-	///Whether array permits transception; can be disabled temporarily by anti-overload measures, or toggled manually
-	var/primed = TRUE
-	///Whether array is currently transceiving
+	///Whether array is currently transceiving (interfacing with a pad for the process of sending or receiving a thing)
 	var/is_transceiving = FALSE
-	///Beam overlay
+	///Beam overlay (this was made an object overlay for the purpose of having access to flick)
 	var/obj/overlay/telebeam
 
 	///Determines if failsafe threshold is equipment power threshold plus transception cost (true) or transception cost (false).
-	var/equipment_failsafe = TRUE
-	///While failsafe is active, communications capability is retained but cargo transception is unavailable. Prompts attempt_restart periodically.
-	var/failsafe_active = FALSE
+	var/use_standard_failsafe = TRUE
+	///Whether array permits transception (false means just comms); disabled by the failsafe when power gets too low
+	var/primed = TRUE
 	///List of items to forcibly send to pads when possible
 	var/list/atom/movable/direct_queue = list()
 
+	///Internal capacitor; cell installed inside the array itself. Draws from grid surplus when available, configurable from the array computer.
+	var/obj/item/cell/intcap = null
+	///Whether the array's conditions for refilling its internal capacitor are satisfied; used for load logic and overlay control
+	var/intcap_charging = FALSE
+	///Whether the door for the internal capacitor's compartment is open
+	var/intcap_door_open = FALSE
+	///How fast the internal capacitor will attempt to draw down grid power while intcap_charging is true
+	var/intcap_draw_rate = 10 KILO WATTS
+	///Amount of surplus past intcap_draw_rate that's required for charging, as a safeguard against spikes in demand
+	var/grid_surplus_threshold = 20 KILO WATTS
+
+	//If the internal capacitor is sabotaged, it will rupture, damaging the capacitor cabinet and bringing the array offline.
+	///This condition tracks the progress of array repair; status of 8 (defined above, change if process changes) indicates full condition
+	var/repair_status = TRSC_FULLREPAIR
```

Comment:
Can you add defines for all the repair levels? Seems like they all have unique repair needs and it would make the exact numbers easier to keep track of.

Reply:
wait also what is this acronym. "Transception Repair Status ??????"

Reply:
Defines added, with a clearer abbreviation.